modules/whoisquery/provider.go

```diff
@@ -0,0 +1,48 @@
+package whoisquery
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/likexian/whois-go"
+	whoisparser "github.com/likexian/whois-parser-go"
+)
+
+type provider interface {
+	remainingTime() (float64, error)
+}
+
+type fromNet struct {
+	domainAddress    string
+}
+
+func newProvider(config Config) (provider, error) {
+	sourceDomain := string(config.Source)
+
+	return &fromNet{domainAddress: sourceDomain}, nil
+}
+
+func (f fromNet) remainingTime() (float64, error) {
+	
+	raw, err := whois.Whois(f.domainAddress)
+
+	if err == nil {
```

Comment:
make the flow (return early if there is fail scenarion)

``go
if err != nil {
  ...
}

...
```